python/cuml/cuml/linear_model/base.pyx

```diff
@@ -73,23 +72,39 @@ class LinearPredictMixin:
                 "LinearModel.predict() cannot be called before fit(). "
                 "Please fit the model first."
             )
-        self.dtype = self.coef_.dtype
-        if len(self.coef_.shape) == 2 and self.coef_.shape[0] > 1:
-            # Handle multi-target prediction in Python.
-            coef_arr = CumlArray.from_input(self.coef_).to_output('array')
-            X_arr = CumlArray.from_input(
-                X,
-                check_dtype=self.dtype,
-                convert_to_dtype=(self.dtype if convert_dtype else None),
-                check_cols=self.n_features_in_
-            ).to_output('array')
-            intercept_arr = CumlArray.from_input(
-                self.intercept_
+
+        multi_target = (len(self.coef_.shape) == 2 and self.coef_.shape[0] > 1)
+        if multi_target:
+            return self._predict_multi_target(X, convert_dtype)
+        else:
+            return self._predict_single_target(X, convert_dtype)
+
+    def _predict_multi_target(self, X, convert_dtype=True) -> CumlArray:
+        """
+        Predict for multi-target case.
+        """
+        coef_arr = CumlArray.from_input(self.coef_).to_output('array')
+        X_arr = CumlArray.from_input(
+            X,
+            check_dtype=self.dtype,
+            convert_to_dtype=(self.dtype if convert_dtype else None),
+            check_cols=self.n_features_in_
+        ).to_output('array')
+        if isinstance(self.intercept_, (int, float, np.number)):
+            intercept_ = self.intercept_
+        else:
+            intercept_ = CumlArray.from_input(
+                self.intercept_,
+                convert_to_dtype=self.dtype if isinstance(self.intercept_, float) else False
```

Comment:
```suggestion
                convert_to_dtype=(self.dtype if convert_dtype else None),
                check_rows=self.n_features_in_
```

Reply:
I've dropped that entirely since there should be no need to convert the dtype at all.